src/Microsoft.ML.TimeSeries/SrCnnAnomalyDetectionBase.cs

```diff
@@ -0,0 +1,298 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.ML.Data;
+using Microsoft.ML.Internal.Utilities;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.Transforms.TimeSeries
+{
+    public class SrCnnAnomalyDetectionBase : IStatefulTransformer, ICanSaveModel
```

Comment:
>SrCnnAnomalyDetectionBase [](start = 17, length = 25)

Out of curiosity, do you plan to have more Anomaly detectors based on this class, or it's gonna be only SRCNN?
If later, I don't see much value in having this base class.

Also would be nice to add some comments to this public class.